File: tests/fixtures/payloads/polyglot/converter.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    char* data;
    size_t length;
} ConversionResult;

typedef struct {
    int timeout;
    char* api_key;
} ConverterConfig;

ConversionResult* convert_document(const char* path, ConverterConfig* config) {
    printf("Converting document in C: %s\n", path);
    
    if (path == NULL) return NULL;
    
    ConversionResult* result = (ConversionResult*)malloc(sizeof(ConversionResult));
    result->data = strdup("c_binary_data");
    result->length = 13;
    
    return result;
}

void free_result(ConversionResult* result) {
    if (result) {
        free(result->data);
        free(result);
    }
}

int main() {
    printf("Polyglot C Test Asset\n");
    return 0;
}

File: tests/fixtures/payloads/polyglot/converter.go
package main

import (
	"context"
	"fmt"
	"path"
)

// DocumentConverter defines the interface for document transformation
type DocumentConverter interface {
	Convert(ctx context.Context, inputPath string) (*ConversionResult, error)
}

type ConversionResult struct {
	Data     []byte
	Metadata map[string]string
}

// PdfConverter implements DocumentConverter for PDF files
type PdfConverter struct {
	Config map[string]interface{}
}

func (p *PdfConverter) Convert(ctx context.Context, inputPath string) (*ConversionResult, error) {
	fmt.Printf("Converting PDF: %s\n", inputPath)
	if path.Ext(inputPath) != ".pdf" {
		return nil, fmt.Errorf("invalid file type")
	}
	return &ConversionResult{
		Data: []byte("pdf data"),
		Metadata: map[string]string{"type": "pdf"},
	}, nil
}

func main() {
	fmt.Println("Polyglot Go Test Asset")
}
